pkg/utils/utils.go

```diff
@@ -70,3 +73,49 @@ func DecompressGZIP(compressedData []byte) (decompressedData []byte, err error)
 
 	return decompressedData, nil
 }
+
+func HumanReadableSize(data []byte) string {
+	const unit = 1024
+	size := len(data)
+	if size < unit {
+		return fmt.Sprintf("%d B", size)
+	}
+	div, exp := int64(unit), 0
+	for n := size / unit; n >= unit; n /= unit {
+		div *= unit
+		exp++
+	}
+	return fmt.Sprintf("%.1f %cB", float64(size)/float64(div), "KMGTPE"[exp])
+}
+
+// TODO: check if it's worth trying to compress the data prior to compressing it
+// This would save some CPU usage client side
+// TODO: have telemetry data type be passed in as a parameter to further check if we should compress data and which algorithm to use
+func ShouldCompress(data []byte) (resultData []byte, compression *string, err error) {
+	// 'compression' is inserted as a sql.NullString, hence it is returned as a nullable string
+	var nilStr *string = nil
+	var validStr string = "gzip"
+
+	compressedData, err := CompressGZIP(data)
```

Comment:
I talked to Fergal about this. It is the plan for future changes. I added a TO DO comment to keep track of it as we move forward. I could add the switch cases to get the code ready for the changes, even though we will only have the default case for now, what do you think?

Reply:
So there are two compression scenarios we are dealing with here...

The over the wire data compression for the telemetry report submissions which comes into play when the telemetry client sends a telemetry report to the server, and in that case we definitely should be thinking in terms of have a switch statement structure to allow for different options that the client may want to use, though for now we can settle for only implementing the gzip case when submitting telemetry reports...

However there is a second compression scenario that we want to address, which is independent of the over the wire scenario and is purely client side, and has nothing to do with the server side compression support, and that is the compression of telemetry data at rest within the client side data store. In the long run we may choose to support multiple compression techniques, which may be based upon the telemetry type or heuristic rules, but initially gzip should be a good starting point as a balance between CPU cost and compression cost... bzip2 is more CPU expensive for slightly better compression for certain data stream patterns, and xz is much more expensive for better compression gains.

When a telemetry source provides us with a telemetry blob, we will want to store that in the client data store in a compressed format, to minimise the amount of disk space consumed by the telemetry client. When we retrieve that telemetry blob from the client data store to send it over the wire we will decompress it so that it can be included uncompressed in the generated telemetry report, maximising the potential compression gain acheieved for the over the wire compression of the telemetry report in those cases where a telemetry report contains multiple similar data items.

Reply:
Reopened issue #12 to further improve on this.
Also ported the changes regarding this matter to a different [PR](https://github.com/SUSE/telemetry/pull/29).
Updates regarding compression will be moved to the new PR as well.